### PR TITLE
Embedded media

### PR DESCRIPTION
**Pull request by wimkeir**
Hey @jerith, hope you're keeping well :)

I'm interning at the Stellenbosch office this holiday and at the moment we're working on extending the Vumi Twitter transport to support embedded images in tweets (issue #20). I figured that it would be easiest to implement the functionality in this client library to keep the transport's code concise.

If you could give my additions a review that would be much appreciated. I've tried to make sure that there are sufficient tests for everything I've done, and to do that I had to extend your existing test helpers slightly - I tried to keep the style and feel consistent as best I could. I've also tested it out in the console and it works as it should.

Let me know what you think :)

### CODE REVIEW

**Comment by wimkeir** [comment]
Looks like Travis is failing because Twisted has dropped support for Python 2.6 :/

**Review by jerith — COMMENTED** [review]
Thanks for doing this work!

I apologise for not reviewing this earlier. The github email must have gotten lost in my spamfilter or something.

I only have one thing to add to the inline comments:
The new `FakeTwitterClient._upload_media()` method (and the `dispatch_multipart()` method it calls) need tests.  You can add them to `TestFakeTwitterClient`, which only contains the tests necessary to exercise the things `FakeTwitterClient` overrides. That will make coveralls happy and then we should be good to merge and release. :-)

**Comment by wimkeir** [comment]
@jerith I've made some changes and Travis/Coveralls seem happy, ready for review :)

File: txtwitter/tests/fake_twitter.py [hunk 1]
```diff
@@ -244,6 +257,33 @@ def to_dict(self, twitter_data):
...
+    def __init__(self, media_id_str, fake_image, **kw):
+        self.media_id_str = media_id_str
+        self.size = fake_image.size
+        self.expires_after_secs = 60
+        self.image = {
+            'image_type': 'image/jpeg',
+            'w': fake_image.width,
+            'h': fake_image.height,
+        }
+
```
Comment (wimkeir) [inline txtwitter/tests/fake_twitter.py]: My thought was that, as long as we're only uploading images, we can expect the response from the Twitter API to be consistent each time, but I guess that's a little short sighted considering we'd like to keep using these entities if the client ever supports uploading of other media types. I'll keep the `kw` rather.

**Review by jerith — APPROVED** [review]
:+1: :cake: :1st_place_medal:

**Comment by jerith** [comment]
@wimkeir Merged and released: https://pypi.python.org/pypi/txTwitter/0.1.6

**Pull request merged**